### PR TITLE
#184651050 Configure the App to use PostgreSQL as DB Engine and Sequelize ORM

### PR DESCRIPTION
**Pull request by ishimwe5555**
### What does this PR do?
Ensures the app uses PostgreSQL as DB Engine, Sequelize ORM and creates a demo user using migrations

### Description of Task to be completed?
-Configure the App to use PostgreSQL as DB
-Set up Sequelize for Object-Relational Mapping(ORM)
-Create a migration for Users table and a seed with default data for testing purposes

### How should this be manually tested?
After cloning the repository and installing all required dependencies, run the following commands to add the Users table to the database and insert a seed(test data):

npx sequelize-cli db:migrate
npx sequelize-cli db:seed:all

### Any background context you want to provide?
I used babel to transpile the migrations and seeders to use ES syntax

###What are the relevant pivotal tracker/Trello stories?
#184651050

### CODE REVIEW

File: src/database/config/config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  production: {
+    username: process.env.POSTGRES_USER,
+    password: process.env.POSTGRES_PASSWORD,
+    database: process.env.POSTGRES_DB,
+    host: process.env.POSTGRES_HOST,
+    dialect: process.env.POSTGRES_DIALECT,
+    port: process.env.POSTGRES_PORT,
+  },
+};
```
Comment (munezerobagira) [inline src/database/config/config.js:38]:
```suggestion
require('dotenv/config')
module.exports = {
  development: {
    url: process.env.DEV_DATABASE_URL,
    dialect: 'postgres',
  },
  test: {
    url: process.env.TEST_DATABASE_URL,
    dialect: 'postgres',
  },
  production: {
    url: process.env.DATABASE_URL,
    dialect: 'postgres',
  },

```
For simplicity let use this format for the environment variables
### connection string format
`postgresql://username:password@host:port/dbname[?param1=value1&...]`

File: src/database/models/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const basename = path.basename(__filename);
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
```
Comment (munezerobagira) [inline src/database/models/index.js:17]:
```suggestion
if (config[env].url) {
  sequelize = new Sequelize(config[env].url);
} else {
  sequelize = new Sequelize(process.env.DEV_DATABASE_URL);
}
```

**Review by munezerobagira — CHANGES_REQUESTED** [review]
Let use the connection string, you cab check the request changes

File: src/database/config/config.js (hunk 1 above)
Comment (ishimwe5555) [reply src/database/config/config.js]: Thanks, the suggestion has applied.

File: src/database/models/index.js (hunk 2 above)
Comment (ishimwe5555) [reply src/database/models/index.js]: Thanks for the suggestion, It's applied now.

**Review by munezerobagira — APPROVED** [review]

**Review by Kress2000 — APPROVED** [review]

**Pull request merged**